mqtt/src/main/java/com/hazelcast/jet/contrib/mqtt/MqttSourceBuilder.java

```diff
@@ -118,6 +119,29 @@
         return this;
     }
 
+    /**
+     * Set that the client and broker should remember state across
+     * restarts and reconnects. The state is persisted to the
+     * {@link com.hazelcast.map.IMap} with the specified name.
+     * <p>
+     * For example:
+     * <pre>{@code
+     * builder.persistSession("mqtt-source-map")
+     * }</pre>
+     * <p>
+     * By default the client and broker will not remember the state. If
+     * this parameter is set, the source ignores the value of
+     * {@link MqttConnectOptions#isCleanSession()} specified via
+     * {@link #connectOptionsFn(SupplierEx)}.
+     * <p>
+     * See {@link MqttConnectOptions#setCleanSession(boolean)}.
+     */
+    @Nonnull
+    public MqttSourceBuilder<T> persistSession(@Nonnull String mapName) {
```

Comment:
Couldn't we generate the map name automatically? The user needs to destroy the map after the job is done, but it's not easy to do.